apps/ejabberd/src/mod_keystore.erl

```diff
@@ -121,7 +121,11 @@ create_keystore_ets() ->
             ok
     end.
 
-maybe_add_heir(EjdSupPid, _Self, BaseOpts) when is_pid(EjdSupPid) ->
+%% In tests or when module is started in run-time, we need to set heir to the
+%% ETS table, otherwise it will be destroy when the creator's process finishes.
+%% When started normally during node start up, self() =:= EjdSupPid and there
+%% is no need for setting heir
+maybe_add_heir(EjdSupPid, EjdSupPid, BaseOpts) ->
     [{heir, EjdSupPid, testing} | BaseOpts];
```

Comment:
You are 100% right. Too many tasks in the same time, too less cores in my brain... I'll fix that. Thanks for careful review!